diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -642,6 +642,57 @@ std::vector<SyntaxDifference> DifferentialFunctionComparator::findAsmDifference(
     return {diff};
 }
 
+/// Implement comparison of global values that does not use a
+/// GlobalNumberState object, since that approach does not fit the use case
+/// of comparing functions in two different modules.
+int DifferentialFunctionComparator::cmpGlobalValues(GlobalValue *L,
+                                                    GlobalValue *R) const {
+    int Result = L <= R;
+
+    if (L->hasName() && R->hasName() && Result) {
+        // Both values are named, compare them by names
+        auto NameL = L->getName();
+        auto NameR = R->getName();
+
+        // Remove number suffixes
+        if (hasSuffix(NameL))
+            NameL = NameL.substr(0, NameL.find_last_of("."));
+        if (hasSuffix(NameR))
+            NameR = NameR.substr(0, NameR.find_last_of("."));
+
+        if (NameL == NameR || (isPrintFunction(NameL) &&
+                isPrintFunction(NameR)))
+            Result = 0;
+    }
+
+    if (isa<GlobalVariable>(L) && isa<GlobalVariable>(R) && Result) {
```

Comment:
I don't think switching the order of the operations will help, since the name comparison will still happen and set the result to zero; not removing the suffix in the case of strings would fix this.

Reply:
The function could start by checking that we are comparing two global constants and compare their initializers in such case (which is what this block does). Otherwise, it should continue by comparing the names (without suffices) and potentially running the function comparator.

Reply:
In other words, if we are comparing global constants, we don't care about the names, just about the values.

Reply:
That will also fix cases when they are two constants with the same name, but different value (they would previously compare as equal although they are in all cases non-equal).